framework/validators/Validator.php

```diff
@@ -231,6 +231,18 @@ public function init()
         $this->attributes = (array) $this->attributes;
         $this->on = (array) $this->on;
         $this->except = (array) $this->except;
+
+        foreach (array_filter($this->initDefaultMessages())
+            as $property => $message
+        ) {
+            if (null === $this->$property) {
+                /**
+                 *  @todo think on how to support custon categories for each
+                 *  message
+                 */
+                $this->$property = Yii::t('yii', $message);
```

Comment:
Yii has message extraction tool which gets strings from code. It works only with `Yii::t('yii', 'My message')`. It can't work with `Yii::t('yii', $message)` because it doesn't evaluate variables.